product_gtin/product_view.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+
+<record id="product_template_search_view" model="ir.ui.view">
+    <field name="model">product.template</field>
+    <field name="inherit_id" ref="product.product_template_search_view"/>
+    <field name="arch" type="xml">
+        <field name="name" position="replace">
```

Comment:
Instead of replacing, you can use position="attributes" and modify only filter_domain attribute.